### PR TITLE
Fdtd fixes

### PR DESCRIPTION
**Pull request by gomezzz**
# Description

Summary of changes

* Small bugfixes / improvements
* Added transient removal
* Refactored calculate_transmission_reflection_coefficients
* Added notebook to compare with TMM and FDTD vs NIDN

Need to 
- [x] fix tests :)

## Resolved Issues

N/A

## How Has This Been Tested?

- [x] Ran notebook
- [x] Ran CI

## Related Pull Requests

N/A

### CODE REVIEW

File: nidn/plots/plot_spectrum.py [hunk 1]
```diff
@@ -100,7 +107,7 @@ def plot_spectrum(
         target_frequencies,
         R_spectrum,
         ylimits,
-        121,
```
Comment (torbjornstoro) [inline nidn/plots/plot_spectrum.py:103]: What is this number?

File: nidn/fdtd_integration/calculate_transmission_reflection_coefficients.py [hunk 2]
```diff
@@ -38,15 +208,14 @@ def calculate_transmission_reflection_coefficients(
                 .item()
             ]
         )
-
     else:
```
Comment (torbjornstoro) [inline nidn/fdtd_integration/calculate_transmission_reflection_coefficients.py:211]: May want to remove this, as the results could possibly be way off, and rather ask that the user uses more time steps? Will probably be unnecessary with the automatic calculation of niter

**Review by torbjornstoro — APPROVED** [review]
Looks good, can't spot anything that would break the fdtd/make results wrong.

File: nidn/fdtd_integration/calculate_transmission_reflection_coefficients.py (hunk 2 above)
Comment (gomezzz) [reply nidn/fdtd_integration/calculate_transmission_reflection_coefficients.py:211]: But we already asking them to increase the number of timesteps, right? Regarding automatic calculation see also https://github.com/esa/NIDN/pull/111/files#r935248046 . I think we should tell the user to do it. Our calculation may be off or the user is just testing something, wants to use fewer timesteps some reason etc. Always forcibly increasing it limits the use options

**Pull request merged**